addons/google_gmail/models/google_gmail_mixin.py

```diff
@@ -72,7 +72,7 @@ def open_google_gmail_uri(self):
         if not self.google_gmail_uri:
             raise UserError(_('Please configure your Gmail credentials.'))
 
-        if not self.smtp_user:
+        if "smtp_user" in self._fields and not self.smtp_user:
```

Comment:
can we overwrite the method in ir.mail_server instead ?

Reply:
This happens when we click the "Connect your Outlook Account"/"Connect your Gmail Account". So unless mistaken it doesn't go through ir.mail_server at this stage (call stack goes through `call_button` on `fetchmail.server`)? But not 100% sure, as I might be missing context.

Reply:
yes, so we remove that check in the mixin, and we do it only in the overwrite

it should be ok no ?

Reply:
Ahhh. So you mean adding the check for example in the `_check_use_microsoft_outlook_service(self)` (ir_mail_server.py in `microsoft_outlook`) method and raise the UserError there?